src/modules/paymentProcessor/StreamingPaymentProcessor.sol

```diff
@@ -656,51 +665,55 @@ contract StreamingPaymentProcessor is Module, IStreamingPaymentProcessor {
             abi.encodeWithSelector(
                 IERC20(_token).transferFrom.selector,
                 client,
-                contributor,
+                paymentReceiver,
                 amount
             )
         );
 
         if (success && (data.length == 0 || abi.decode(data, (bool)))) {
-            emit TokensReleased(contributor, _token, amount);
+            emit TokensReleased(paymentReceiver, _token, amount);
+
+            //Make sure to let paymentClient know that amount doesnt have to be stored anymore
+            IERC20PaymentClient(client).amountPaid(amount);
```

Comment:
1: Yeah youre right here. Will change it accordingly
2: This would definitely be a nice to have. Due to some time constraints and a bit of mismatched planning I would like to implement this at a later point of time.